### PR TITLE
bazel/linux: fix kernel_module to define its runfiles

### PR DESCRIPTION
**Pull request by george-enf**
kernel_module should define its runfiles so that the following works:

kernel_module(name = "mod" ...)
sh_binary(name = "test", srcs = ["test.sh"], data = [":mod"])

Jira: https://enfabrica.atlassian.net/browse/SF-219
Signed-off-by: George Prekas <george@enfabrica.net>

### CODE REVIEW

File: bazel/linux/defs.bzl [hunk 1]
```diff
@@ -126,7 +126,15 @@ def _kernel_modules(ctx):
             setup = ctx.attr.setup,
         ))
 
-    return [DefaultInfo(files = depset(outputs)), KernelBundleInfo(modules = bundled)]
+    return [
+        DefaultInfo(
+            files = depset(outputs),
+            runfiles = ctx.runfiles(files = outputs),
```
Comment (ccontavalli) [inline bazel/linux/defs.bzl:132]: is this a generic rule? Eg, if a binary depends on a target for data, only runfiles are brought in, and must be declared as runfiles? What triggers the requirement for those files to be runfiles?

(eg, should we recommend to do this for every rule we write going forward? Why is it necessary here, but not in other places?)

**Review by ccontavalli — APPROVED** [review]
a question below, but fine to merge as is.

File: bazel/linux/defs.bzl (hunk 1 above)
Comment (rajachan) [reply bazel/linux/defs.bzl:132]: Both outputs and runfiles should be available as data for sh_binary targets from what I've seen. Maybe are outputs are not being enumerated properly with the kernel rules?

Comment (george-enf) [reply bazel/linux/defs.bzl:132]: I created a very simple test case:
BUILD.bazel
```
load("simple_rule.bzl", "simple_rule")
simple_rule(
    name = "simple_rule1",
)
sh_binary(
    name = "test1",
    srcs = ["test.sh"],
    data = [":simple_rule1"],
)
```
test.sh
```
find
```
simple_rule.bzl
```
def _simple_rule(ctx):
    outputs = []
    for i in range(1, 3):
        output = ctx.actions.declare_file("file" + str(i))
        outputs.append(output)
        ctx.actions.write(output, "foo" + str(i))
    return [
        DefaultInfo(
            files = depset(outputs),
            # runfiles = ctx.runfiles(files = outputs),
        ),
    ]
simple_rule = rule(
    implementation = _simple_rule,
)
```
And then, I do `bazel run test1`. I can only see `file1` and `file2` if I uncomment the `runfiles` above. It looks to me that `runfiles` can be a disjoint set to `files`.

**Review by ccontavalli — APPROVED** [review]

**Pull request merged**